vuepress/docs/docs/stitching.md

```diff
@@ -0,0 +1,202 @@
+  } yield {
+    val remoteResolvers = remoteSchemaResolvers.resolvers
+
+    // 3
+    implicit val githubProfileSchema: Schema[ZEnv, Repository] =
+      remoteSchemaResolvers
+        .remoteResolver("Repository")(
+          // 4
+          // Here we need to translate our local `Repository` case class into
+          // a top-level query which can be issued towards Github's API.
+          // We do this by accepting a `caliban.execution.Field`, representing
+          // all the selected fields for a repository and map that to the
+          // top-level `repository` query in the Github API.
+          // This means the final query will end up looking something like this:
+          // query {
+          //   repository(owner: r.args.owner, name: r.args.name) {
+          // .   <incoming query>
+          //   }
+          // }
+          RemoteResolver.fromFunction((r: ResolveRequest[Repository]) =>
+            r.field.copy(
+              name = "repository",
+              arguments = Map(
+                "owner" -> Value.StringValue(r.args.owner),
+                "name"  -> Value.StringValue(r.args.name)
+              )
+            )
+          ) >>> remoteResolvers.default
+        )
+        .provide(sttpClient)
+  }
+```
+
+However, when running this we will experience failing requests due to `401 Unauthorized`. This is because all queries to Github's API requires authorization to be provided. In order to fix this, we need to add authorization to both the introspection query as well as our remote resolver. We also need a config module that can provide us with a Github token based on the value of `GITHUB_TOKEN` in our environment.
+
+```scala
+case class Configuration(githubToken: String)
+
+object Configuration {
+  def fromEnvironment =
+    (for {
+      githubToken <- read("GITHUB_TOKEN")
+    } yield Configuration(githubToken)).toLayer
+
+  private def read(key: String): Task[String] = Task.effect(
+    sys.env(key)
+  )
+}
+```
+
+We can now update the introspection query to use our token:
+
+```scala
+  val api = for {
+    config     <- ZIO.environment[Has[Configuration]]
+    schemaLoader = SchemaLoader.fromIntrospection(
+                     GITHUB_API,
+                     Some(
+                       List(
+                         Options.Header(
+                           "Authorization",
+                           s"Bearer ${config.get.githubToken}"
+                         )
+                       )
+                     )
+                   )
+    // ...
+  }  yield ???
+```
+
+as well as update our resolver to authorize our request.
+
+In order to do this, we can use `RemoteResolver[R, E, A, B]` which lets us compose resolution steps via `>>>`.
+
+In order to make this code easier, we can extract the mechanics around sending the actual request:
+
+```scala
+val apiRequest =
+    remoteResolvers.toQuery >>> remoteResolvers.request >>> RemoteResolver.fromEffect((r: HttpRequest) =>
```

Comment:
Oh I see you move the `apiURL` into the class constructor so that you don't have to call it here? I guess you could ignore my comments about the `private` method then.

Reply:
I changed this according to your suggestions, see above!